api/src/pcapi/core/educational/api/dms.py

```diff
@@ -0,0 +1,165 @@
+    db.session.commit()
+
+    logger.info(
+        "[DMS] Finishing processing procedure %s.",
+        procedure_number,
+        extra={
+            "procedure_number": procedure_number,
+        },
+    )
+
+
+def _get_or_create_application(procedure_number: int, node: dict) -> None:
+    data = {
+        "state": node["state"],
+        "procedure": procedure_number,
+        "application": node["number"],
+        "siret": node["demandeur"].get("siret"),
+        "lastChangeDate": _convert_iso_string_to_naive_utc_datetime(node["dateDerniereModification"]),
+        "depositDate": _convert_iso_string_to_naive_utc_datetime(node["dateDepot"]),
+        "expirationDate": _convert_iso_string_to_naive_utc_datetime(node["dateExpiration"]),
+        "buildDate": _convert_iso_string_to_naive_utc_datetime(node["datePassageEnConstruction"]),
+        "instructionDate": _convert_iso_string_to_naive_utc_datetime(node["datePassageEnInstruction"]),
+        "processingDate": _convert_iso_string_to_naive_utc_datetime(node["dateTraitement"]),
+        "userDeletionDate": _convert_iso_string_to_naive_utc_datetime(node["dateSuppressionParUsager"]),
+    }
+
+    if not data["siret"]:
+        logger.info(
+            "[DMS] Application parsing failed: no siret found",
+            extra={
+                "application_number": data["application"],
+                "application_scalar_id": node.get("id"),
+                "procedure_number": procedure_number,
+            },
+        )
+        return
+
+    application = educational_models.CollectiveDmsApplication.query.filter(
+        educational_models.CollectiveDmsApplication.procedure == procedure_number,
+        educational_models.CollectiveDmsApplication.siret == data["siret"],
+    ).one_or_none()
+    if application:
+        for key, value in data.items():
+            setattr(application, key, value)
+    else:
+        venue = offerers_models.Venue.query.filter(offerers_models.Venue.siret == data["siret"]).one_or_none()
+        if not venue:
+            logger.info(
+                "[DMS] Application parsing failed: unknown siret",
+                extra={
+                    "application_number": data["application"],
+                    "application_scalar_id": node.get("id"),
+                    "procedure_number": procedure_number,
+                },
+            )
+            return
+        application = educational_models.CollectiveDmsApplication(venue=venue, **data)
+        db.session.add(application)
+    db.session.commit()
+
+
+def _convert_iso_string_to_naive_utc_datetime(date_str: str | None) -> datetime | None:
+    if not date_str:
+        return None
+    date = datetime.fromisoformat(date_str)
+    # dms always returns an iso date with timezone therefore utcoffset never returns None
+    offset = typing.cast(timedelta, date.utcoffset())
+    utc_date = date - offset
+    naive_date = utc_date.replace(tzinfo=None)
+    return naive_date
```

Comment:
(détail) J'ai fait un rapide test et il me semble que ça marcherait comme suit :  

```suggestion
    return date.astimezone(pytz.utc).replace(tzinfo=None) 
```